spec/views/notifications/index.html.erb_spec.rb

```diff
@@ -1,69 +1,78 @@
+  let(:patch_note_type_a) { create(:patch_note_type, name: "patch_note_type_a") }
+  let(:patch_note_type_b) { create(:patch_note_type, name: "patch_note_type_b") }
+  let(:patch_note_1) { create(:patch_note, note: "Patch Note 1", patch_note_type: patch_note_type_a) }
+  let(:patch_note_2) { create(:patch_note, note: "Patch Note B", patch_note_type: patch_note_type_b) }
+
   before do
-    travel_to Date.new(2021, 1, 1)
+    assign(:notifications, Noticed::Notification.all)
+    assign(:patch_notes, PatchNote.all)
+    assign(:deploy_time, deploy_time)
   end
 
   context "when there is a deploy date" do
-    let(:notification_created_after_deploy_a) { create(:notification) }
-    let(:notification_created_after_deploy_b) { create(:notification) }
-    let(:notification_created_at_deploy) { create(:notification) }
-    let(:notification_created_before_deploy_a) { create(:notification) }
-    let(:notification_created_before_deploy_b) { create(:notification) }
+    let(:deploy_time) { 2.days.ago }
 
     before do
-      Health.instance.update_attribute(:latest_deploy_time, 2.days.ago)
+      Health.instance.update_attribute(:latest_deploy_time, deploy_time)
     end
 
     context "when there are notifications" do
       before do
-        # TODO invalid notifications that break the view
-        # notification_created_after_deploy_a.update_attribute(:created_at, 1.hour.ago)
-        # notification_created_after_deploy_b.update_attribute(:created_at, 1.day.ago)
-        # notification_created_at_deploy.update_attribute(:created_at, 2.days.ago)
-        # notification_created_before_deploy_a.update_attribute(:created_at, 2.days.ago - 1.hour)
-        # notification_created_before_deploy_b.update_attribute(:created_at, 3.days.ago)
+        notification_1_hour_ago.update_attribute(:created_at, 1.hour.ago)
+        notification_1_day_ago.update_attribute(:created_at, 1.day.ago)
+        notification_2_days_ago.update_attribute(:created_at, 2.days.ago)
+        notification_3_days_ago.update_attribute(:created_at, 3.days.ago)
+
+        patch_note_1.update_attribute(:patch_note_group, patch_note_group_all_users)
       end
 
-      xit "has all notifications created after and including the deploy date above the patch note" do
-        # TODO fill in after notification factory is given ability to create a notification with notes
+      it "has all notifications created after and including the deploy date above the patch note" do
+        render template: "notifications/index"
+
+        notifications_html = Nokogiri::HTML5(rendered).css(".list-group-item")
+        patch_note_index = notifications_html.index { |node| node.text.include?("Patch Notes") }
+
+        expect(notifications_html[0].text).to include(notification_1_hour_ago.event.message)
+        expect(notifications_html[1].text).to include(notification_1_day_ago.event.message)
+        expect(notifications_html[2].text).to include(notification_2_days_ago.event.message)
```

Comment:
Can we hard code these? `expect(notifications_html[0].text).to include("Message 0")`